### PR TITLE
Update nnp_training to have identical setup to NC

### LINKED ISSUE #243: Make torchani's training setup identical to neurochem
After a thorough comparison between Neurochem and Torchani, I have found a few differences as explained briefly in the following:

- [x] NC skips the weight decay for the first layer of each atomic network, even if the weight decay parameter is provided in the input. This was the main source of difference in training performance.(#275)
- [x] In NC the weights are initialized to He Kaiming normal and the biases to zero. (#245, #275)
- [x] Loss in NC is scaled by sqrt(natoms). (#275)
- [x] NC uses AdamW optimizer to update the weights and SGD for the biases. (#275)
- [x] There were a couple of discrepancies between theory and the implementation of AdamW in Torchani (#261).
- [x] NC scale the gradients of atomic networks by the number of atoms in that batch, and compute the gradient from SumSquaredError instead of MeanSquaredError. (will be fixed in Neurochem)
- [ ] NC reshuffle the batches after every epoch (tracked in #248, no much improvement as tested, low priority)

### PR DESCRIPTION
**Pull request by farhadrgh**
This PR fixes #243 

Make `nnp_training.py` setup identical to Neurochem.
- Fixed docs, removed warnings
- SGD added
- Removed 1st layer's weight decay parameters

### CODE REVIEW

File: examples/nnp_training.py [hunk 1]
```diff
@@ -193,53 +195,58 @@ def init_params(m):
...
     {'params': [O_network[4].bias]},
-    {'params': O_network[6].parameters()},
-])
+    {'params': [O_network[6].bias]},
+], lr=1e-3)
 
 ###############################################################################
 # Setting up a learning rate scheduler to do learning rate decay
-scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, factor=0.5, patience=100, threshold=0)
+scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(AdamW, factor=0.5, patience=100, threshold=0)
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Do we need LR decay for both AdamW and SGD? Same for the NC trainer

File: examples/nnp_training.py [hunk 2]
```diff
@@ -254,8 +261,8 @@ def init_params(m):
 # Resume training from previously saved checkpoints:
 if os.path.isfile(latest_checkpoint):
     checkpoint = torch.load(latest_checkpoint)
-    model.load_state_dict(checkpoint['nn'])
-    optimizer.load_state_dict(checkpoint['optimizer'])
+    nn.load_state_dict(checkpoint['nn'])
+    AdamW.load_state_dict(checkpoint['optimizer'])
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Should there also be a SGD loader and saver?

File: examples/nnp_training.py [hunk 3]
```diff
@@ -306,41 +313,51 @@ def validate():
...
         predicted_energies = []
         num_atoms = []
+
         for chunk_species, chunk_coordinates in batch_x:
             num_atoms.append((chunk_species >= 0).to(true_energies.dtype).sum(dim=1))
             _, chunk_energies = model((chunk_species, chunk_coordinates))
             predicted_energies.append(chunk_energies)
-        num_atoms = torch.cat(num_atoms)
+
+        num_atoms = torch.cat(num_atoms).to(true_energies.dtype)
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Do we need the `.to(true_energies.dtype)`?

Comment (farhadrgh) [reply examples/nnp_training.py]: No, fixed it in https://github.com/aiqm/torchani/pull/280/commits/cc8664a44b2e37cd8476a570fd605f65c615d274
